rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/SymbolTable.kt

```diff
@@ -100,8 +100,30 @@ class SymbolTable : ISymbolTable {
         require(data.type.canBeAssigned(value)) {
             "Value $value cannot be assigned to data: $data"
         }
-        names[data.name.uppercase()] = data
-        return values.put(data, value.forType(data.type))
+
+        if (data is FieldDefinition) {
+            val containerValue = get(data.container)
+            if (data.container.isArray()) {
+                throw IllegalStateException("We do not yet handle an array container")
+            } else if (data.declaredArrayInLine != null) {
+                ProjectedArrayValue.forData(containerValue as DataStructValue, data).container.set(data, value.forType(data.type))
+                return ProjectedArrayValue.forData(containerValue as DataStructValue, data)
+            } else {
+                // Should be always a DataStructValue
+                if (containerValue is DataStructValue) {
+                    containerValue.set(data, value.forType(data.type))
```

Comment:
I don't understand why in `DataStructValue` you put an instance of `Value` returned by `value.forType` and after that you return the coercion of that instance.

Reply:
// step 1: the following statement set the plain value into the DS
containerValue.set(data, value.forType(data.type))

// step 2: the following statement, get plain value from the DS and coerces it to return the real type of data
return coerce(containerValue[data], data.type)

I could have returned the following value:
return value.forType(data.type)

but if something went wrong during the step 1, I could have passed the wrong value back to the caller.
In this way, I am sure that I am returning the real value from the DS.

Let me know your thoughts

Reply:
The matter is that we are handing the value related `DataDefinition` different by the `FieldDefinition` and I don't know what is the right way, see below:
![image](https://github.com/smeup/jariko/assets/40103274/46a2bfcf-38db-4703-befc-93f5c54fadfb)
However, if it works, perhaps it is not useful to ask too many questions...

Reply:
I understand your doubts, but this way to get and store values from DSs has been already used in other part of Jariko.
I think that they forgot to implement it everywhere. I just did the same thing as you can see from the image below. 

![image](https://github.com/smeup/jariko/assets/32836304/324d48c3-374f-467f-90b8-15c2bdd666ea)

Reply:
Ok